epictrack-web/src/components/workPlan/event/EventList.tsx

```diff
@@ -324,11 +318,25 @@ const EventList = () => {
     setShowTemplateForm(false);
     setShowMilestoneForm(false);
     getCombinedEvents();
-    getTemplateUploadStatus();
     getWorkPhases();
-    getWorkById();
     setTaskEvent(undefined);
     setMilestoneEvent(undefined);
+    getTemplateUploadStatus()
+      .then(() => {
+        return getWorkById();
+      })
+      .then(() => {
+        if (
+          milestoneEvent &&
+          milestoneEvent.event_configuration.event_position ===

```

Comment:
You should add one more condition to make sure user has actually completed the phase by checking milestoneEvent.actual_date.